crates/solvers/src/boundary/naive.rs

```diff
@@ -108,6 +113,15 @@ pub fn solve(
     let boundary_solution =
         multi_order_solver::solve(&slippage.context(), boundary_orders, &boundary_pool)?;
 
+    let nmb_orders = boundary_solution.trades().count();
+    let gas = INITIALIZATION_COST
+        + SETTLEMENT
+        + nmb_orders as u64 * (2 * ERC20_TRANSFER + liquidity.gas.0.as_u64()); // this is pessimistic in case the pool is not used
```

Comment:
Naive solvers only use the liquidity source once (no matter how many orders are in the batch):


```suggestion
        + (nmb_orders as u64 * 2 * ERC20_TRANSFER)
        + liquidity.gas.0.as_u64()); // this is pessimistic in case the pool is not used
```

Reply:
Fixed.